internal/queryset/methods/queryset.go

```diff
@@ -194,6 +194,34 @@ func NewNotInFilterMethod(ctx QsFieldContext) InFilterMethod {
 	return newInFilterMethodImpl(ctx, "NotIn", "NOT IN")
 }
 
+// InFilterMethod filters with IN condition
+type FunctionOneArgumentMethodFilterMethod struct {
+	chainedQuerySetMethod
+	onFieldMethod
+	nArgsMethod
+	qsCallGormMethod
+}
+
+func newFunctionOneArgumentMethodFilterMethod(ctx QsFieldContext, operationName, functionName string) FunctionOneArgumentMethodFilterMethod {
+	ctx = ctx.WithOperationName(operationName)
+	argName := fieldNameToArgName(ctx.fieldName())
+	args := newNArgsMethod(
+		newOneArgMethod(argName, "..."+ctx.fieldTypeName()),
+	)
+	return FunctionOneArgumentMethodFilterMethod{
+		onFieldMethod:         ctx.onFieldMethod(),
+		nArgsMethod:           args,
+		chainedQuerySetMethod: ctx.chainedQuerySetMethod(),
+		qsCallGormMethod: newQsCallGormMethod("Where", "\"%s(%s, ?)\", %s",
+			functionName, ctx.fieldDBName(), argName),
```

Comment:
is this sql injection safe?

Reply:
It generates code exactly like any other code generation we are doing. This code is not running in production
[5:32](https://valencesecurity.slack.com/archives/D029JEL828Y/p1663597929107829)
for example the new output we have is:
// AliasesJsonbExists is an autogenerated method
// nolint: dupl
func (qs PrincipalAccountQuerySet) AliasesJsonbExists(aliases ...string) PrincipalAccountQuerySet {
	return qs.w(qs.db.Where("JSONB_EXISTS(aliases, ?)", aliases))
}
[5:32](https://valencesecurity.slack.com/archives/D029JEL828Y/p1663597960023109)
I don’t think anyone can compromise this more than what we already have:
// ConnectorUidIn is an autogenerated method
// nolint: dupl
func (qs PrincipalAccountQuerySet) ConnectorUidIn(connectorUid ...string) PrincipalAccountQuerySet {
	if len(connectorUid) == 0 {
		qs.db.AddError(errors.New("must at least pass one connectorUid in ConnectorUidIn"))
		return qs.w(qs.db)
	}
	return qs.w(qs.db.Where("connector_uid IN (?)", connectorUid))
}
[5:34](https://valencesecurity.slack.com/archives/D029JEL828Y/p1663598074296229)
does that answer you question?


Shlomi
  [5:35 PM](https://valencesecurity.slack.com/archives/D029JEL828Y/p1663598142224359)
yes, actually. that’s what the “?” is for. approving.